src/MessagingClient.ts

```diff
@@ -235,7 +236,11 @@ export class MessagingClient implements MessagingAPI {
         const roomMessages = room.timeline.filter(event => event.getType() === EventType.RoomMessage)
```

Comment:
Is it possible to `roomMessages` to be empty after the filter?

In that case `roomMessages[roomMessages.length - 1]` would be OOB, should we add the check?

Reply:
good catch!! indeed, as far as I remember, this is possible because the explorer always marks the chat as read when the chat is opened, and the timeline window might also be empty of chat messages.